### PR TITLE
Mals 189 low power functions doc

### PR DESCRIPTION
**Pull request by amtrang**
### Linked Jira Issue:
https://ser401-2022-group39.atlassian.net/jira/software/projects/MALS/issues/MALS-189

### Description of Changes:
Create a document describing the new low-power mode functionality.

### Acceptance Criteria:
- [x] AC1: A single document describing all added low-power mode features has been created
- [x] AC2: Document has been placed in the DxCore repository

### CODE REVIEW

File: megaavr/extras/LowPower.md [hunk 1]
```diff
@@ -0,0 +1,635 @@
+# LowPower.cpp
+    This document is a comprehensive list of all the functions from LowPower.cpp. Each function includes descriptions for how they change the hardware to decrease the power draw for DxCore microcontrollers.
+    All information for the hardware changes and interrupts can be found in the AVR128DA48 datasheet from Microchip which can be found here: https://www.microchip.com/en-us/product/AVR128DA48
```
Comment (jnbrauer) [inline megaavr/extras/LowPower.md:3]: Not quite sure why but GitHub is rendering all of the indented sections as code blocks, let's fix that so these sections display as paragraphs. 
```suggestion
This document is a comprehensive list of all the functions from LowPower.cpp. Each function includes descriptions for how they change the hardware to decrease the power draw for DxCore microcontrollers.
All information for the hardware changes and interrupts can be found in the AVR128DA48 datasheet from Microchip which can be found here: https://www.microchip.com/en-us/product/AVR128DA48
```

File: megaavr/extras/LowPower.md [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+
+## setSleepMode(SLPCTRL_SMODE_t sleepMode)
+Description:
```
Comment (jnbrauer) [inline megaavr/extras/LowPower.md:6]: These labels could be either level 3 headers or bolded to make them more distinct.

File: megaavr/extras/LowPower.md [hunk 3]
```diff
@@ -0,0 +1,635 @@
+# LowPower.cpp
```
Comment (jnbrauer) [inline megaavr/extras/LowPower.md:1]: Let's add a paragraph or two at the start of this document describing the features of the low-power library and its goals at a high level.

File: megaavr/extras/LowPower.md [hunk 4]
```diff
@@ -0,0 +1,635 @@
...
+    All information for the hardware changes and interrupts can be found in the AVR128DA48 datasheet from Microchip which can be found here: https://www.microchip.com/en-us/product/AVR128DA48
+
+## setSleepMode(SLPCTRL_SMODE_t sleepMode)
+Description: 
+
+    Sets the microcontroller's specified sleep mode, but does not enter sleep mode automatically.
+
+Arguments: 
+    
+    sleepMode - the type of sleep mode the microcontroller will use when it enters sleep mode.
```
Comment (jnbrauer) [inline megaavr/extras/LowPower.md:12]: I would suggest using code tags for function arguments, names, and sample calls throughout. 

```suggestion
`sleepMode` - the type of sleep mode the microcontroller will use when it enters sleep mode.
```

**Review by jnbrauer — CHANGES_REQUESTED** [review]

**Review by cgjeffries — CHANGES_REQUESTED** [review]
Content looks fine at a glance, but I think formatting could be improved.
Perhaps we could re-use/repurpose the content from the API design doc we created in the beginning? https://github.com/jnbrauer/SER401-2022-Group39-Documentation/blob/main/low_power_api_design.md

**Review by jnbrauer — APPROVED** [review]

**Pull request merged**